instance/models/server.py

```diff
@@ -419,30 +324,35 @@ def start(self):
         TODO: Create the key dynamically
         """
         self.logger.info('Starting server (status=%s)...', self.status)
-        self._transition(self._status_to_started)
-        os_server = openstack.create_server(
-            self.nova,
-            self.instance.sub_domain,
-            settings.OPENSTACK_SANDBOX_FLAVOR,
-            settings.OPENSTACK_SANDBOX_BASE_IMAGE,
-            key_name=settings.OPENSTACK_SANDBOX_SSH_KEYNAME,
-        )
-        self.openstack_id = os_server.id
-        self.logger.info('Server got assigned OpenStack id %s', self.openstack_id)
-        self._set_progress(Progress.Success, expected_status=Status.Started)
+        self._transition(self._status_to_building)
+        try:
+            os_server = openstack.create_server(
+                self.nova,
+                self.instance.sub_domain,
+                settings.OPENSTACK_SANDBOX_FLAVOR,
+                settings.OPENSTACK_SANDBOX_BASE_IMAGE,
+                key_name=settings.OPENSTACK_SANDBOX_SSH_KEYNAME,
+            )
+        except novaclient.exceptions.ClientException:
```

Comment:
Would this cover "quota limitations & waiting list" as mentioned in the docstring above? If so, it's a partial implementation of that TODO item :)